### PR TITLE
Upgrade jsonrpc to upgrade hyper

### LINKED ISSUE #803: cargo audit shows vulnerability in tendermint-light-node (hyper v0.12.35)
**Steps to reproduce**

```
> cargo audit

...

Crate:         hyper
Version:       0.12.35
Title:         Multiple Transfer-Encoding headers misinterprets request payload
Date:          2021-02-05
ID:            RUSTSEC-2021-0020
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0020
Solution:      Upgrade to >=0.14.3 OR >=0.13.10, <0.14.0
Dependency tree: 
hyper 0.12.35

...
```

It seems like this is due to the dependency on `jsonrpc-http-server` in the Light Node:

```
> cargo tree -i hyper:0.12.35
hyper v0.12.35
└── jsonrpc-http-server v14.2.0
    └── tendermint-light-node v0.18.0
```

**What's the definition of "done" for this issue?**

When `cargo audit` gives us a clean bill of health.

### PR DESCRIPTION
**Pull request by thanethomson**
Closes #803 

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

File: tendermint/src/lite.rs [hunk 1]
```diff
@@ -1,6 +0,0 @@
-//! Core logic and traits of a light client.
```
Comment (thanethomson) [inline tendermint/src/lite.rs:1]: This file isn't used by any modules in the tendermint crate.

Comment (xla) [reply tendermint/src/lite.rs:1]: :clap:

**Review by xla — APPROVED** [review]
🚥 🗽 🌌 ⛴

**Review by romac — DISMISSED** [review]

**Review by romac — APPROVED** [review]

**Review by xla — APPROVED** [review]
🌗 🎞 😀 🎎

**Pull request merged**